ISISComputingGroup/ibex_gui
# Ticket1124 device screens list

###Description of work

Screen to show device screen list from PV. When devices are clicked an OPI is opened.

###To test

For ticket https://github.com/ISISComputingGroup/IBEX/issues/1124
Also see: https://github.com/ISISComputingGroup/EPICS-inst_servers/pull/54
System tests: https://github.com/ISISComputingGroup/ibex_system_tests/pull/19
###Acceptance criteria

1. Device are displayed in the list which appear in the device screen xml.
    1. XML can be edited by adding it to text box (white area).
    1, Example xml below
1. Device screen updates when PVs update
1. Device screen  changes with different instruments
1. Device screen can be sorted by clicking on titles
1. Device screen has appropriate icons for various devices (key is opi resource key)
1. When item is clicked OPI is shown
    1. OPI is correct opi
    1. Macros are set
    1. Tab has the correct title
1. Multiple devices can be shown
1. When item is clicked for a second time OPI is brought to the front
1. OPI can be closed
1. When instrument is switched open OPIs are closed.
1. Check synoptic OPIs still work.

Example XML
```
<?xml version="1.0" ?>
<devices xmlns="http://epics.isis.rl.ac.uk/schema/screens/1.0/">
	<device>
		<name>My Kepco</name>
		<key>Kepco</key>
		<type>OPI</type>
		<properties>
			<property>
				<key>IOC_NUM</key>
				<value>01</value>
			</property>
			<property>
				<key>Property2</key>
				<value>Value2</value>
			</property>
		</properties>
	</device>
	<device>
		<name>Attenuator</name>
		<key>Attenuator</key>
		<type>OPI</type>
		<properties>
			<property>
				<key>Q</key>
				<value>ATTEN</value>
			</property>
		</properties>
	</device>
	<device>
		<name>Attenuator B</name>
		<key>Attenuator</key>
		<type>OPI</type>
		<properties>
			<property>
				<key>Q</key>
				<value>ATTEN</value>
			</property>
		</properties>
	</device>	
	<device>
		<name>Error IOC</name>
		<key>MyDeviceKey</key>
		<type>COMPONENT</type>
		<properties>
			<property>
				<key>Property1</key>
				<value>Value1</value>
			</property>
			<property>
				<key>Property2</key>
				<value>Value2</value>
			</property>
		</properties>
	</device>
</devices>
```

---

#### Code Review


### Functional Tests
